In [1]:
import pandas as pd
#Read the thanksgiving.csv file. This file isn't encoded normaly so, I specify "Latin-1" for encoding
data = pd.read_csv("thanksgiving.csv",encoding = "Latin-1")
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


In [2]:
#Display the column names of the dataframe
data.columns

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

In [3]:
data['Do you celebrate Thanksgiving?'].value_counts()

# Filter out rows in data where the responses of the questin "Do you celebrate Thanksgiving?" is No
data= data[data['Do you celebrate Thanksgiving?']== 'Yes']

In [4]:
# Explore the values in the column "What is typically the main dish at your Thanksgiving dinner?"
data['What is typically the main dish at your Thanksgiving dinner?'].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [5]:
# Display the column 'Do you typically have gravy?' where the dish in the column 'What is typically the main dish at your Thanksgiving dinner' is Tofurkey
data[data['What is typically the main dish at your Thanksgiving dinner?']== "Tofurkey"]['Do you typically have gravy?']

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

In [6]:
# Figure out how many people eat Pies, and what Pies people eat
apple_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple'])
pumpkin_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin'])
pecan_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan'])

ate_pies = (apple_isnull) & (pumpkin_isnull) & (pecan_isnull)
ate_pies.value_counts()


False    876
True     104
dtype: int64

The ate_pies series display False in 876 values. Therefore this is the number of people that ate pies in Thanksgiving day.

In [7]:
# In this step, I extract the first item of the column Age. Example: column Age = 18 - 29, the value will be 18. 
# Remove the + character from 60+ and convert the column Age to numeric(int)
def convert_age(row):
    age = row['Age']
    age_is_null = pd.isnull(age)
    if age_is_null == True:
        return None
    age_split = age.split( )[0]
    age_remove = age_split.rstrip("+")
    age_int = int(age_remove)
    return age_int
data['int_age'] = data.apply(convert_age,axis=1)


    

In [8]:
data['int_age'].describe()

/dataquest/system/env/python3/lib/python3.4/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%             NaN
50%             NaN
75%             NaN
max       60.000000
Name: int_age, dtype: float64

This metodology has the disadvantage that take the first number of the interval. If the age of the respondent are between 30 - 44, the age will be 30, and if the respondent is older than 60, the function assign a age of 60. 

I think that is important to take into account the withdraw of this method because assign people less age than they have. 

In [9]:
def convert_income(row):
    income = row['How much total combined money did all members of your HOUSEHOLD earn last year?']
    income_null = pd.isnull(income)
    if income_null == True:
        return None
    income_split = income.split( )[0]
    if income_split == 'Prefer':
        return None
    income_clean = income_split.replace("$","")
    income_clean_all = income_clean.replace(",","")
    income_int = int(income_clean_all)
    return income_int
data['int_income'] = data.apply(convert_income,axis=1)

In [10]:
data['int_income'].describe()

/dataquest/system/env/python3/lib/python3.4/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%                NaN
50%                NaN
75%                NaN
max      200000.000000
Name: int_income, dtype: float64

The results of the int_income function, take the first item of the category in the value of the column 'How much total combined money did all members of your HOUSEHOLD earn last year?'. 
This could introduce a downward bias in the result, because the methodology assign to the respondent, the lower value of the range category. The people that are within the first range($0 to $9,999) will have an income of 0. 


In [11]:
# Observe if there are correlation between Travel Distance and Income
# First, I analyse the people who earn under 50000 and the correlation with the travel column
data[data['int_income'] < 50000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         106
Thanksgiving is local--it will take place in the town I live in                      92
Thanksgiving is out of town but not too far--it's a drive of a few hours or less     64
Thanksgiving is out of town and far away--I have to drive several hours or fly       16
Name: How far will you travel for Thanksgiving?, dtype: int64

In [12]:
data[data['int_income'] < 150000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         281
Thanksgiving is local--it will take place in the town I live in                     203
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    150
Thanksgiving is out of town and far away--I have to drive several hours or fly       55
Name: How far will you travel for Thanksgiving?, dtype: int64

The results show that people who earn more than 150000 are more expected to celebrate Thanksgiving day at their home or in the town they live. 

In [13]:
# In the next two steps, I link friendship with age and income. For this I make 2 pivot tables to analyze these two variables with the Friendsgiving event.
data.pivot_table(index='Have you ever tried to meet up with hometown friends on Thanksgiving night?',columns='Have you ever attended a "Friendsgiving?"',values = 'int_age')

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [14]:
data.pivot_table(index='Have you ever tried to meet up with hometown friends on Thanksgiving night?',columns='Have you ever attended a "Friendsgiving?"',values = 'int_income')

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

The pivot table that relate the age with the Friendsgiving event, show that young people are more likely to celebrate the event with friends. People who have an average age of 34, respondent Yes, for attending a Friendsgiving and for meet up with hometown friends on Thanksgiving night. 

The second table, show that pepole with less income in average are more expected to celebrate Friendgiving, and to meet up with hometown friend on Thanksgiving night. These two questions are related with friendship. 

We can conclude that friendship, is most common between young people and people with less income in average. 

In [21]:
# To figure out the most common dessert people eat, I joined all the columns that ask about the dessert people eat in th Thanksgiving dinner with the concat method,
# and then utilize the value_counts method to the whole series.
apples = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Apple cobbler']
blondies = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Blondies']
brownies = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Brownies']
carrot_cake = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Carrot cake']
cheese_cake = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake']
cookies = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cookies']
fudge = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Fudge']
ice_cream = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream']
peach_cobbler = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Peach cobbler']
none = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None']
other = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Other (please specify)']
other_1 = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Other (please specify).1']
dessert = pd.concat([apples,blondies,brownies,carrot_cake,cheese_cake,cookies,fudge,ice_cream,peach_cobbler,none,other,other_1],axis=0)
dessert.value_counts()

None                                                           295
Ice cream                                                      266
Cookies                                                        204
Cheesecake                                                     191
Other (please specify)                                         134
Brownies                                                       128
Apple cobbler                                                  110
Peach cobbler                                                  103
Carrot cake                                                     72
Fudge                                                           43
Blondies                                                        16
pie                                                             13
Pie                                                             12
pies                                                             6
Pumpkin pie                                                   

In [25]:

Us_region = data.groupby('US Region')['What is typically the main dish at your Thanksgiving dinner?'].value_counts()
Us_region

US Region           What is typically the main dish at your Thanksgiving dinner?
East North Central  Turkey                                                          135
                    Other (please specify)                                            5
                    Ham/Pork                                                          4
                    Tofurkey                                                          1
East South Central  Turkey                                                           50
                    Other (please specify)                                            4
                    Ham/Pork                                                          1
                    Roast beef                                                        1
Middle Atlantic     Turkey                                                          130
                    Tofurkey                                                          5
                    Other (please speci

In [19]:
# An interesting point in this project, is to figure out the most common complete meal people eat.
# A way of perform this is to count all the items that compose the complete meal and see the results.
# These items are main dish, side dishes, pie, dessert. The main dish just got the result, as well as the dessert.
# Then I will perform this task to side dishes and pie
# Value counts of side dishes
Brussel_sprouts = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts']
Carrots = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots']
Cauliflower = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Cauliflower']
Corn = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn']
Cornbread = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Cornbread']
Fruit_salad = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Fruit salad']
Green_beans_green_bean_casserole = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole']
Macaroni_cheese = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Macaroni and cheese']
Mashed_potatoes = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes']
Rolls_biscuits = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Rolls/biscuits']
Squash = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Squash']
Vegetable_salad = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Vegetable salad']
Yams_sweet_potato_casserole = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Yams/sweet potato casserole']
Other_side_dish = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify)']
Other_side_dish_1 = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify).1']

side_dishes = pd.concat([Brussel_sprouts,Carrots,Cauliflower,Corn,
Cornbread,Fruit_salad,Green_beans_green_bean_casserole,Macaroni_cheese,
Mashed_potatoes,Rolls_biscuits,Squash,Vegetable_salad,Yams_sweet_potato_casserole,
Other_side_dish, Other_side_dish_1], axis=0)

side_dishes.value_counts()


Mashed potatoes                                                 817
Rolls/biscuits                                                  766
Green beans/green bean casserole                                686
Yams/sweet potato casserole                                     631
Corn                                                            464
Carrots                                                         242
Cornbread                                                       235
Fruit salad                                                     215
Vegetable salad                                                 209
Macaroni and cheese                                             206
Squash                                                          171
Brussel sprouts                                                 155
Other (please specify)                                          111
Cauliflower                                                      88
broccoli                                        

In [38]:
# Value counts of Pie
Apple = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple']
Buttermilk = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk']
Cherry = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry']
Chocolate = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Chocolate']
Coconut_cream = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Coconut cream']
Key_lime = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Key lime']
Peach = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Peach']
Pecan = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan']
Pumpkin = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin']
Sweet_Potato = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Sweet Potato']
No_apply = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - None']
Other = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify)']
Other_1 = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify).1']

pie = pd.concat([Apple,Buttermilk,Cherry,Chocolate,Coconut_cream,Key_lime,
Peach,Pecan,Pumpkin,Sweet_Potato,No_apply, Other, Other_1],axis=0)
counts = pie.value_counts()
pie = pd.DataFrame({"Pie": counts.index, 'number': counts})
pie

Pie  \
Pumpkin                                                                                Pumpkin   
Apple                                                                                    Apple   
Pecan                                                                                    Pecan   
Sweet Potato                                                                      Sweet Potato   
Chocolate                                                                            Chocolate   
Cherry                                                                                  Cherry   
Other (please specify)                                                  Other (please specify)   
None                                                                                      None   
Key lime                                                                              Key lime   
Coconut cream                                                                    Coconut cream   
Buttermilk                                                                          Buttermilk   
Peach                                                                                    Peach   
Chess                                                                                    Chess   
minced meat                                                                        minced meat   
mince meat                                                                          mince meat   
Lemon                                                                                    Lemon   
mincemeat                                                                            mincemeat   
Mincemeat                                                                           Mincemeat    
peanut butter                                                                    peanut butter   
Blueberry or Peach Cobbler                                          Blueberry or Peach Cobbler   
Mincemeat                                                                            Mincemeat   
Derby, Japanese fruit                                                    Derby, Japanese fruit   
Shoo Fly                                                                              Shoo Fly   
We serve pine nut cake not pie                                  We serve pine nut cake not pie   
lemon merang                                                                      lemon merang   
I don't know                                                                      I don't know   
rhubarb                                                                                rhubarb   
Concord grape                                                                    Concord grape   
Strawberry cream                                                              Strawberry cream   
custard                                                                                custard   
...                                                                                        ...   
berry                                                                                    berry   
raspberry                                                                            raspberry   
ice box pie                                                                        ice box pie   
pecan                                                                                    pecan   
Shoo Fly Pie                                                                     Shoo Fly Pie    
banana cream                                                                      banana cream   
Cocunut custard                                                                Cocunut custard   
Mince meat                                                                          Mince meat   
Pumpkin chiffon                                                                Pumpkin chiffon   
blueberry                                                                

Analyzing the value counts of the items in the meal, it concludes that Mashed Potatoes ,Pumpkin and Ice Creams are the most side dishes , pie and dessert respectevely that people eat for Thanksgiving day. 
Also, there is a lot number of people that don't eat dessert, because the None value for this was 295.

In [57]:
import numpy as np
data.pivot_table(index = 'US Region', values = ['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Chocolate','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Sweet Potato','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Rolls/biscuits','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Yams/sweet potato casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots'] , aggfunc= 'count')

Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots  \
US Region                                                                                                                                 
East North Central                                                 28                                                                     
East South Central                                                 15                                                                     
Middle Atlantic                                                    45                                                                     
Mountain                                                           11                                                                     
New England                                                        25                                                                     
Pacific                                                            30                                                                     
South Atlantic                                                     48                                                                     
West North Central                                                 12                                                                     
West South Central                                                 21                                                                     

                    Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn  \
US Region                                                                                                                              
East North Central                                                 76                                                                  
East South Central                                                 31                                                                  
Middle Atlantic                                                    77                                                                  
Mountain                                                           17                                                                  
New England                                                        22                                                                  
Pacific                                                            55                                                                  
South Atlantic                                                     96                                                                  
West North Central                                                 36                                                                  
West South Central                                                 46                                                                  

                    Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole  \
US Region                                                                                                                                                          
East North Central                                                103                                                                                              
East South Central                                                 49                                                                                              
Middle Atlantic                                                    92                                                                                              
Mountain                                                           31                                                                                              
New Engla